### ISSUE TITLE
Missing appender for post types without `core/paragraph` support

### ISSUE THREAD

**Issue by chrisvanpatten** [WordPress/gutenberg#32891]
## Description
If a post type does not have support for the default block (`core/paragraph`), a button appender should be visible:

https://github.com/WordPress/gutenberg/blob/77bc9cae4d016f52523f0ee7018183961ab6056e/packages/block-editor/src/components/block-list-appender/index.js#L72-L79

<img width="1093" alt="Screen Shot 2021-06-22 at 9 17 43 AM" src="https://user-images.githubusercontent.com/1231306/122931263-b6995c80-d33a-11eb-88e9-74bd8a6c0d17.png">

However, since Gutenberg 9.9.0, the button is not visible. However, it _is_ in the DOM, and you can hover it and view the tooltip:

<img width="1112" alt="Screen Shot 2021-06-22 at 9 05 10 AM" src="https://user-images.githubusercontent.com/1231306/122931400-dc266600-d33a-11eb-9628-d16ff01ee05e.png">

<img width="746" alt="Screen Shot 2021-06-22 at 9 21 38 AM" src="https://user-images.githubusercontent.com/1231306/122931860-43dcb100-d33b-11eb-9b95-12c7d98b0416.png">

I am still combing through the commits but it _seems_ unlikely that this is intentional, given that the button is still in the DOM and doesn't appear to be an affordance for keyboard users.

## Step-by-step reproduction instructions

Add the following to your site:

```php
add_filter( 'allowed_block_types', function () {
    return [ 'core/image' ]; // for example
} );
```

Create a new post or page.

## Expected behaviour
You should see a button appender.

## Actual behaviour
Observe there is a visible appender in Gutenberg 9.8.4, but not in Gutenberg 9.9.0.

## WordPress information
- WordPress version: latest
- Gutenberg version: 9.9.0 through latest

**Comment by chrisvanpatten** [comment]
@jasmussen I'm pretty sure I've tracked this down to #28219, and specifically the addition of a `width` — I left a comment on the PR as well.

Not quite sure how to resolve without breaking the intention behind the change in that PR though 🙃

**Comment by jasmussen** [comment]
Thank you for such an excellent ticket. I have a fix in #32894
